libc-test/build.rs

```diff
@@ -381,6 +381,7 @@ fn test_openbsd(target: &str) {
         "limits.h",
         "link.h",
         "locale.h",
+        "machine/spinlock.h",
```

Comment:
this definition shouldn't be need now that you don't use `_atomic_lock_t` type.

the tests passes on OpenBSD without it.